SPEC.md

```diff
@@ -4586,31 +4618,58 @@ Example output:
 </p>
 </details>
 
-##### `maxCpu`
+##### `max_cpu`
+
+* Accepted types:
+    * `Int`
+    * `Float`
+* Alias: `maxCpu`
 
 Specifies the maximum CPU to be provisioned for a task. The value of this hint has the same specification as [`requirements.cpu`](#cpu).
```

Comment:
I think there needs to be some guidance around what max means (or could mean depending on the engine). For example if I set a max CPU (or memory) should I expect that hint to result in a (probably hard) limit on a container or cgroup or should I expect that this means that the compute instance that gets the task may have no more than X cpu or memory (which might lead to conflicts in the ability to provision the minimums). If we leave it completely up to the engines to decide then different interpretations by different engines would reduce or remove any possible value of the hint.

It's fine to say that the hint can be ignored but if it isn't what should a user expect?

Reply:
In this case I think the hint is really just meant as a potential optimization. I.e. it's communicating to the engine that the task will use no more than `maxCpu` cpus, so it is free to limit the provisioned resources. But maybe others have a different view.

Reply:
Could we reword "_Specifies the maximum CPU to be provisioned for a task_" to something like "_A hint to the engine that the task expects to use no more than maxCpu. The engine is free to make any optimization it deems beneficial based on this information_".

Reply:
I updated the language for all the hints using your first sentence as a template. The second sentence I instead put as a general statement at the top level of the hints section.